shared/js/background/components/resource-loader.js

```diff
@@ -168,8 +168,9 @@ export default class ResourceLoader extends EventTarget {
      *  contents: any;
      *  etag?: string;
      * }} result
+     * @param {boolean} waitForUpdateProcessing
      */
-    async _updateData (result) {
+    async _updateData (result, waitForUpdateProcessing = false) {
```

Comment:
I did that initially but I found it caused a deadlock. I didn't run it through a debugger to see exactly why, for now it seemed simpler to avoid waiting by default.